Automattic/wp-calypso
# My Home: Add "Remind me" button to "Connect accounts" task

#### Changes proposed in this Pull Request

This PR adds the ability to skip the "Connect accounts" task with a "Remind me" feature that will will hide the card for the specified time.

<img width="981" alt="Screen Shot 2020-04-22 at 16 40 01" src="https://user-images.githubusercontent.com/1233880/79996044-441fb900-84b8-11ea-9c4f-a9392a87c21f.png">

Since this is a feature that will be needed by all "single tasks" we're planning to implement, and all of them share the same structure, this PR also adds a new `Task` component in order to make easier the development of future tasks.

#### Testing instructions

* Apply D42273-code and sandbox the API.
* Load the experimental Home layout by going to `/home/:site?flags=home/experimental-layout` for an established site.
* Enable the analytics debug by typing `localStorage.debug='calypso:analytics*'` in the browser console and reloading My Home.
* Make sure the "Connect accounts" task shows up.
* Make sure there is a "Remind me" dropdown displaying the "Tomorrow", "Next week" and "Never" options.
* Select one of them and quickly restore the task by undoing the action (you can only do this during the 5 seconds the notice is visible).
* Make sure the card disappears, a success notice is displayed with an "Undo" button and clicking on it restores the task back.
* Make sure stats are tracked for both actions (skip and restore):
<img width="941" alt="Screen Shot 2020-04-23 at 12 53 33" src="https://user-images.githubusercontent.com/1233880/80102115-a3420400-8572-11ea-9ab8-274498bff67d.png">

* Reload the experimental Home layout.
* Make sure the task is still there.
* Dismiss the task again but don't restore it this time.
* Make sure the card disappears.
* Reload the experimental Home layout.
* Make sure the card doesn't show up.
* Simulate the selected time has passed so it's time to display again the card:
  * In D42273-code, manually update the threshold to 0 for the selected reminder option. 
  * For example, if you chose "Tomorrow", make sure the threshold for the `1d` case is 0.
```php
case '1d':
	$threshold = 0;
	break;
```
* Reload the experimental Home layout again.
* Make sure the card shows up now.
